publisher/lib/fibos_collector.py

```diff
@@ -0,0 +1,81 @@
+import argparse
+import os
+
+from rdflib import Graph, OWL, RDF
+
+
+def collect_fibos(fibo_folder_path: str, fibo_dev_file_path: str, fibo_prod_file_path: str):
+    print('Collecting FIBO Dev and Prod')
+
+    fibo_dev = Graph()
+    fibo_prod = Graph()
+
+    print('Getting external ontologies')
+
+    fibo_dev.parse('http://www.w3.org/2002/07/owl')
+    fibo_dev.parse('http://www.w3.org/2000/01/rdf-schema')
+    fibo_dev.parse('http://www.w3.org/1999/02/22-rdf-syntax-ns')
+    fibo_dev.parse('https://www.omg.org/spec/LCC/Countries/CountryRepresentation/')
+    fibo_dev.parse('https://www.omg.org/spec/LCC/Languages/LanguageRepresentation/')
+
+    fibo_prod.parse('http://www.w3.org/2002/07/owl')
+    fibo_prod.parse('http://www.w3.org/2000/01/rdf-schema')
+    fibo_prod.parse('http://www.w3.org/1999/02/22-rdf-syntax-ns')
+    fibo_prod.parse('https://www.omg.org/spec/LCC/Countries/CountryRepresentation/')
+    fibo_prod.parse('https://www.omg.org/spec/LCC/Languages/LanguageRepresentation/')
+
+    print('Importing FIBO ontologies')
+
+    fibo_prod_ontolog_iri_local_parts = __get_fibo_prod_ontology_iri_local_parts(fibo_folder_path=fibo_folder_path)
+    for root, dirs, files in os.walk(fibo_folder_path):
+        for file in files:
+            with open(os.path.join(root, file), "r") as fibo_file:
+                filename, file_extension = os.path.splitext(file)
+                if 'rdf' in file_extension:
+                    if not 'etc' in root:
+                        ontology = Graph()
+                        print('Importing', file)
+                        try:
+                            ontology.parse(fibo_file)
+                            fibo_dev += ontology
+                            ontology_IRIs = ontology.subjects(predicate=RDF.type, object=OWL.Ontology)
+                            for ontology_IRI in ontology_IRIs:
+                                iri_parts = str(ontology_IRI).split(sep='/')
+                                if len(iri_parts) < 2:
+                                    continue
+                                ontology_IRI_local_part = iri_parts[-2]
+                                if ontology_IRI_local_part in fibo_prod_ontolog_iri_local_parts:
+                                    fibo_prod += ontology
+                                    continue
+                        except Exception as exception:
+                            print('Cannot parse', file, 'because', str(exception.args))
+                fibo_file.close()
+
+    fibo_dev.serialize(destination=fibo_dev_file_path)
+    fibo_prod.serialize(destination=fibo_prod_file_path)
+
+
+def __get_fibo_prod_ontology_iri_local_parts(fibo_folder_path: str) -> list:
+    fibo_prod_ontolog_iri_local_parts = []
+    prod_fibo_file = open(os.path.join(fibo_folder_path, 'AboutFIBOProd.rdf'))
```

Comment:
there are a lot of hard-coded FIBO-related references

Reply:
Is there a pattern I can follow?
I looked into a couple of shell files and found multiple hard-coded FIBO-related references all over the place, e.g., in https://github.com/edmcouncil/ontology-publisher/blob/master/publisher/product/ontology/build.sh.
Similarly for the python scripts.

Reply:
I managed to generalised FIBO references up to the level of Docker, i.e., python script and shell script are FIBO-free or at least are less FIBO-dependent.
I am not able to generalise them further.